src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/CountPrisonersService.kt

```diff
@@ -12,16 +12,17 @@ class CountPrisonersService(
   private val offenderSearchService: OffenderSearchService,
 ) {
   /**
-   * Returns the number of prisoners on given level at a prison
+   * True if there are currently any prisoners on a given incentive level at a prison
    */
-  suspend fun countOfPrisonersOnLevelInPrison(prisonId: String, levelCode: String): Int {
-    return offenderSearchService.findOffendersAtLocation(prisonId)
-      .fold(0) { countSoFar, offenders ->
+  suspend fun prisonersExistOnLevelInPrison(prisonId: String, levelCode: String): Boolean {
+    var prisonersExistOnLevel = false
+    offenderSearchService.findOffendersAtLocation(prisonId)
+      .takeWhile { offenders ->
         val bookingIds = offenders.map { it.bookingId }
-        countSoFar + prisonerIepLevelRepository.findAllByBookingIdInAndCurrentIsTrueOrderByReviewTimeDesc(bookingIds)
-          .count {
-            it.iepCode == levelCode
-          }
+        prisonersExistOnLevel = prisonerIepLevelRepository.somePrisonerCurrentlyOnLevel(bookingIds, levelCode)
+        !prisonersExistOnLevel
       }
+      .collect()
```

Comment:
let me guess, it wouldn't do anything/start without the `collect()`

Reply:
Yes 😝
I got stuck on it until I realised take still returns a flow and flows do nothing unless you pull on them